bot/modals/class_modal.py

```diff
@@ -0,0 +1,164 @@
+from discord.ui import Modal, TextInput
+
+from bot.consts import Colors
+from bot.data.class_repository import ClassRepository
+from bot.messaging.events import Events
+from bot.models.class_models import ClassChannelScaffold
+from bot.sock_bot import SockBot
+from bot.utils.helpers import error_embed
+
+# Min and max of major length, i.e., 'CPSC', 'HCC', 'MATH'
+MIN_MAJOR_LEN = 3
+MAX_MAJOR_LEN = 4
+# Min and max of class title length, i.e., 'Data Structures and Algorithms', etc.
+MIN_TITLE_LEN = 5
+MAX_TITLE_LEN = 50
+# Min and max of class instructor, i.e., 'Dean', 'Widman', 'VanScoy', etc.
+MIN_INSTR_LEN = 2
+MAX_INSTR_LEN = 15
+# Max length of class description, i.e., 'Students learn about...'
+MAX_DESCR_LEN = 500
+# Min and max of class number, i.e., 1060, 4910, 8400, etc.
+MIN_CLASS_NUM = 1000
+MAX_CLASS_NUM = 8999
+
+
+class AddClassModal(Modal):
+
+    major = TextInput(
+        label='Course Major',
+        default='cpsc',
+        placeholder='cpsc',
+        min_length=MIN_MAJOR_LEN,
+        max_length=MAX_MAJOR_LEN,
+        row=0
+    )
+    course_number = TextInput(
+        label='Course Number',
+        placeholder='2120',
+        min_length=4,
+        max_length=4,
+        row=1
+    )
+    course_title = TextInput(
+        label='Course Name',
+        placeholder='Data Structures & Algorithms',
+        min_length=MIN_TITLE_LEN,
+        max_length=MAX_TITLE_LEN,
+        row=2
+    )
+    professor = TextInput(
+        label='Course Instructor',
+        placeholder='Dean',
+        min_length=MIN_INSTR_LEN,
+        max_length=MAX_INSTR_LEN,
+        row=3
+    )
+    course_description = TextInput(
+        label='Course Description',
+        placeholder='Students learn about...',
+        style=TextStyle.long,
+        max_length=MAX_DESCR_LEN,
+        required=False,
+        row=4
+    )
+
+    def __init__(self, bot: SockBot,
+                 *,
+                 class_data: tuple[str, int] | None = None,
+                 channel: discord.TextChannel | None = None,
+                 role: discord.Role | None = None):
+        super().__init__(title='📔 Add Class' if not channel else f'Insert #{channel.name}')
+        self._bot = bot
+        self._role = role
+        self._channel = channel
+        self._repo = ClassRepository()
+        if class_data or channel:
+            self._autofill(channel if channel else class_data)
+
+    async def on_submit(self, inter: Interaction) -> None:
+        # correct for error - reject if invalid
+        if not valid_course_num(self.course_number.value):
+            embed = error_embed(inter.user, f'Course number `{self.course_number.value}` is invalid.')
+            await inter.response.send_message(embed=embed, ephemeral=True)
+            return
+        # format our data, so it's nice and neat and capitalized correctly (I know you guys don't use caps)
+        professor = self.professor.value.split(' ')[-1].capitalize()
```

Comment:
split defaults to a string with one space, remove the unnecessary param

Reply:
Neat, didn't know that